src/services/AuthService.ts

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+import { cache } from 'memory-cache';
+
+var JWT_SECRET = 'my-secret-key';
+
+interface User {
+  id: string;
+  email: string;
+  role: string;
+  lastLogin: Date;
+}
+
+export class AuthService {
+  private static instance: AuthService;
+  private users: any = {};
+
+  public static getInstance(): AuthService {
+    if (!AuthService.instance) {
+      AuthService.instance = new AuthService();
+    }
+    return AuthService.instance;
+  }
+
+  async authenticateUser(email: string, password: string) {
```

Comment:
it looks like in this prototype we are storing everything in memory. can we finalize a decision on how we plan to persist the auth tokens on the backend? as we do that, we probably should augment the password with a SALT secret that we maintain only server side